pkg/preflight/execute.go

```diff
@@ -125,20 +165,6 @@ func execute(appID string, sequence int64, preflightSpec *troubleshootv1beta2.Pr
 		uploadPreflightResults.Results = results
 	}
 
-	logger.Debug("preflight marshalling")
-	b, err := json.Marshal(uploadPreflightResults)
-	if err != nil {
-		return uploadPreflightResults, errors.Wrap(err, "failed to marshal results")
-	}
-
-	completeMx.Lock()
-	defer completeMx.Unlock()
-
-	isComplete = true
-	if err := store.GetStore().SetPreflightResults(appID, sequence, b); err != nil {
```

Comment:
This is the call that updates data and stops polling, but it was only being called when preflights run successfully.